This patch reenables build imgressample. Unlike previous version of this
patch, this uses a more lightweight approach.

With building imgresample, a few symbol clashes occur with libswscale. We
therefore strip off symbols that are already provided by libswscale.

References:

<http://permalink.gmane.org/gmane.comp.video.ffmpeg.devel/69238>
http://svn.mandriva.com/cgi-bin/viewvc.cgi/packages/cooker/ffmpeg/current/SOURCES/ffmpeg-reenable-imgresample.patch?revision=207573&view=markup

--- a/libavcodec/Makefile
+++ b/libavcodec/Makefile
@@ -351,9 +351,7 @@ OBJS-$(HAVE_W32THREADS)                +
 
 OBJS-$(HAVE_XVMC_ACCEL)                += xvmcvideo.o
 
-ifneq ($(CONFIG_SWSCALER),yes)
 OBJS += imgresample.o
-endif
 
 # processor-specific code
 ifeq ($(HAVE_MMX),yes)
--- a/libavcodec/imgconvert.c
+++ b/libavcodec/imgconvert.c
@@ -2324,7 +2324,7 @@ int img_pad(AVPicture *dst, const AVPict
 }
 #endif
 
-#ifndef CONFIG_SWSCALER
+#if 1
 /* XXX: always use linesize. Return -1 if not supported */
 int img_convert(AVPicture *dst, int dst_pix_fmt,
                 const AVPicture *src, int src_pix_fmt,
--- a/libavcodec/imgresample.c
+++ b/libavcodec/imgresample.c
@@ -512,6 +512,12 @@ void img_resample_close(ImgReSampleConte
     av_free(s);
 }
 
+/*
+ * the following symbols are provided by libswscale. Disable them in
+ * libavcodec so that applications can actually link against both
+ * avcodec and libswscale.
+ */
+#if 0
 struct SwsContext *sws_getContext(int srcW, int srcH, int srcFormat,
                                   int dstW, int dstH, int dstFormat,
                                   int flags, SwsFilter *srcFilter,
@@ -688,7 +694,7 @@ the_end:
     av_free(buf2);
     return res;
 }
-
+#endif
 
 #ifdef TEST
 #include <stdio.h>
